opal/models.py

```diff
@@ -722,12 +769,51 @@ def get_display_template(cls, team=None, subteam=None):
                 0, 'records/{0}/{1}.html'.format(team, name))
             if subteam:
                 list_display_templates.insert(
-                    0, 'records/{0}/{1}/{2}.html'.format(team,
-                                                              subteam,
-                                                              name))
+                    0, 'records/{0}/{1}/{2}.html'.format(
+                        team, subteam, name
+                    )
+                )
         return find_template(list_display_templates)
 
     @classmethod
+    def bulk_update_from_dicts(
+        cls, parent, list_of_dicts, user
+    ):
+        """
+            allows the bulk updating of a field for example
+            [
+                {"test": "blah 1", "details": "blah blah"}
+                {"test": "blah 2", "details": "blah blah"}
+            ]
+
+            parent is the parent class, that can be Episode or Patient
+        """
+        schema_name = parent.__class__.__name__.lower()
+
+        if cls._is_singleton:
+            if len(list_of_dicts) > 1:
+                raise ValueError(
+                    "attempted creation of multiple fields on a singleton {}".format(cls.__name__)
+                )
+
+        for a_dict in list_of_dicts:
+            if "id" in a_dict or cls._is_singleton:
+                if cls._is_singleton:
+                    query = cls.objects.filter(**{schema_name: parent})
+
+                    if query.exists():
+                        subrecord = query.get()
+                    else:
+                        subrecord = cls(**{schema_name: parent})
+                else:
+                    subrecord = cls.objects.get(id=a_dict["id"])
+            else:
```

Comment:
This probably needs a test ? 
Maybe I'm missing it, but I can't seem to see anything that updates a previously existing non singleton subrecord?